### PR TITLE
table-reservation api

### PR DESCRIPTION
**Pull request by PavloCherniovyi**

### CODE REVIEW

File: table-reservation/app/controllers/reservations.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      const [ reservationId ] = await makeReservation(time, end, guests, availableTable);
+      if (reservationId) {
+        return res
+          .status(201)
+          .append('Location', link(req, `/api/reservation/${reservationId}`))
+          .send({ message: 'Successfully created' });
+      }
+      res.status(500).send({ message: 'Error while creating reservation' });
+    }
+    res.status(400).send({ message: 'There is no available table' });
```
Comment (webwin0) [inline table-reservation/app/controllers/reservations.js]: 404 status code

File: table-reservation/app/controllers/reservations.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  static async getReservationInfo(req, res) {
+    const { reservation_id } = req.params;
+    const [ reservation ] = await getReservation(reservation_id);
+    if (reservation) return res.status(200).send({ reservation });
+    res.sendStatus(404);
```
Comment (webwin0) [inline table-reservation/app/controllers/reservations.js]: Better to send response in json format for all api responses
res.status(404).send({message: 'Reservation is not found'});

File: table-reservation/app/controllers/reservations.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  static async updateReservation(req, res) {
+    const { reservation_id } = req.params;
+    const { reservation: { time, duration, guests } } = req.body;
+    const [ reservation ] = await getReservation(reservation_id);
+
+    if (reservation) {
+      const wasDeleted = await deleteReservation(reservation.id);
```
Comment (webwin0) [inline table-reservation/app/controllers/reservations.js]: You shouldn't delete an existent reservation in update method

File: table-reservation/app/controllers/reservations.js (hunk 2 above)
Comment (webwin0) [reply table-reservation/app/controllers/reservations.js]: here and in all responses below

File: table-reservation/app/helpers/queries.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  );
+
+const getReservation = async id =>
+  await knex('reservations as r')
+    .join('tables as t', 'r.table_id', '=', 't.id')
+    .select('r.*', 't.number', 't.capacity')
+    .where('r.id', id);
+
+const deleteReservation = async id =>
+  await knex.transaction(trx =>
```
Comment (webwin0) [inline table-reservation/app/helpers/queries.js]: `await` is not necessary in async function before return

File: table-reservation/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+app.disable('x-powered-by');
+app.use(bodyParser.json());
+app.use(cookieParser());
+
+app.use('/api', reservationsRoutes);
+
+app.use((err, req, res, next) => {
+  if (err instanceof ValidationError) {
+    res.status(400).json({ message: { error: err } });
+    return next();
```
Comment (webwin0) [inline table-reservation/index.js]: if you sent response than next middleware doesn't needed

**Comment by PavloCherniovyi** [comment]
@webwin0 Please review. Commented code fixed.

**Pull request merged**